File: python/p010.py
def compute(LIMIT):
	prime = [True] * LIMIT
	i = 2
	while (i * i) <= LIMIT:
		if prime[i] == True:
			j = i * i
			while j <= LIMIT - 1:
				prime[j] = False
				j += i
		i += 1
	sum = 0
	for i in range(2, LIMIT):
		if prime[i] == True:
			sum += i
	return sum

if __name__ == "__main__":
	print(compute(2000000))

File: c/p010.c
#include "stdbool.h"
#include "stdio.h"

long long int compute(const int LIMIT) {
	bool prime[LIMIT];
	for (int i = 0; i < LIMIT; i++)
		prime[i] = true;
	for (long int i = 2; i * i <= LIMIT; i++) {
		if (prime[i] == true) {
			for (long long int j = i * i; j <= LIMIT; j += i)
				prime[j] = false;
		}
	}
	long long int sum = 0;
	for (int i = 2; i <= LIMIT; i++) {
		if (prime[i] == true)
			sum += i;
	}
	return sum;
}

int main() {
	printf("%lli\n", compute(2000000));
	return 0;
}
